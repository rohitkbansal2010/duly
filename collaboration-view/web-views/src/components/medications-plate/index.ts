export { MedicationsPlate } from './medications-plate';
export { MedicationsPlateSkeleton } from './medications-plate-skeleton';
