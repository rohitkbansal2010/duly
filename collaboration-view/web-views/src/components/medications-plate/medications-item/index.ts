export { MedicationsItem } from './medications-item';
