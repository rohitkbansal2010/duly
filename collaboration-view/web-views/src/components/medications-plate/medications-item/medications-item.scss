@import "~@styles/typography.scss";
@import "~@styles/themes.scss";
@import "~../medications-plate.scss";

.medicationsMain {
  &Item {
    display: flex;
    width: 100%;
    border-bottom: 0.0625rem solid $gray;

    &Medicine {
      @include medicine;
      display: flex;
      flex-direction: column;

      padding: 1.25rem;

      &Title {
        @include nexa-heavy-highlight;

        font-size: var(--font-size-18px-4px) !important;
        // line-height: 1.625rem;
      }

      &Reason {
        @include nexa-regular-highlight;
        font-size: var(--font-size-14px-4px) !important;
        // line-height: 1.625rem;
      }

      &Additional {
        font-size: var(--font-size-14px-4px) !important;
        @include inter-text-default;
      }
    }

    &Instructions {
      @include inter-text-default;
      @include instructions;

      padding: 1.25rem;
      font-size: var(--font-size-14px-4px) !important;
      // line-height: 1.375rem;
    }
  }
}
