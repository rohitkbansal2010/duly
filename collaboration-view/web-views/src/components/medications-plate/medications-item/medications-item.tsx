import { MedicationsData } from '@types';
import { formatMDYYYYDate, showFirstPrefixItem } from '@utils';

import styles from './medications-item.scss';

export const MedicationsItem = ({
  title = '',
  reason = '',
  startDate,
  instructions = '',
  provider,
}: MedicationsData) => {
  const prescriber = provider && `${showFirstPrefixItem(provider.prefixes)}${provider.familyName}`.trim();

  return (
    <div className={styles.medicationsMainItem}>
      <div className={styles.medicationsMainItemMedicine}>
        <span className={styles.medicationsMainItemMedicineTitle}>{title}</span>
        <span className={styles.medicationsMainItemMedicineReason}>{reason}</span>
        <span className={styles.medicationsMainItemMedicineAdditional}>
          {startDate && `Started: ${formatMDYYYYDate(startDate)} `}
          {prescriber && `Prescriber: ${prescriber}`}
        </span>
      </div>
      <div className={styles.medicationsMainItemInstructions}>
        {instructions}
      </div>
    </div>
  );};
