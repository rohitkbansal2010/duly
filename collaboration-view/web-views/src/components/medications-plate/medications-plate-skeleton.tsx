import { DulyLoader } from '@components/duly-loader';
import { MEDICATION_LOADER_WIDTH } from '@constants';

import styles from './medications-plate.scss';

export const MedicationsPlateSkeleton = () => 
  (
    <div className={styles.medicationsPlateSkeleton}>
      <DulyLoader width={MEDICATION_LOADER_WIDTH} />
    </div>
  );
