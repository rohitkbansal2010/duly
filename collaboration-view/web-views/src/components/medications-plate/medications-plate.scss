@import "~@styles/typography.scss";
@import "~@styles/themes.scss";

@mixin border-right {
  border-right: 0.0625rem solid $gray;
}

@mixin medicine {
  @include border-right;

  width: 40%;
}

@mixin instructions {
  width: 60%;
}

.medicationsPlate {
  &Container {
    margin-bottom: 1.25rem;
    padding: 1.25rem;
    background: $white;
    border-radius: 0.5rem;

    &:last-child {
      margin-bottom: 0;
    }
  }

  &Main {
    display: flex;
    flex-direction: column;
    background: $light-gray;
    border: 0.0625rem solid $gray;
    border-top: none;
    border-radius: 0 0 0.625rem 0.625rem;

    :last-child {
      border-bottom: none;
    }
  }

  &Skeleton {
    position: relative;
    flex-grow: 1;
    margin-bottom: 1.25rem;
    background: $white;
    border-radius: 0.5rem;

    &:last-child {
      margin-bottom: 0;
    }
  }
}
