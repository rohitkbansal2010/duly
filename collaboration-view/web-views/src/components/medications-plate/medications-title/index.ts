export { MedicationsTitle } from './medications-title';
