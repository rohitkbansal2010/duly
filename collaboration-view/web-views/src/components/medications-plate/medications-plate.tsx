import { MedicationsType } from '@enums';
import { MedicationsData } from '@types';
import { sortMedications } from '@utils';

import { MedicationsHeader } from './medications-header';
import { MedicationsItem } from './medications-item';
import styles from './medications-plate.scss';
import { MedicationsTitle } from './medications-title';

type MedicationsPlateProps = {
  type: MedicationsType;
  medications: MedicationsData[];
}

export const MedicationsPlate = ({ type, medications }: MedicationsPlateProps) => 
  (
    <div className={styles.medicationsPlateContainer}>
      <MedicationsTitle type={type} count={medications.length}/>
      <MedicationsHeader type={type}/>
      <div className={styles.medicationsPlateMain}>
        {sortMedications(medications).map(medication => 
          <MedicationsItem key={medication.id} {...medication}/>)}
      </div>
    </div>
  );
