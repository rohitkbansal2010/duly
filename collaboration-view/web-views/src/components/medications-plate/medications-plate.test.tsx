import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { MedicationsType } from '@enums';
import { mockMedicationsWidgetData } from '@mock-data';

import { MedicationsItem } from './medications-item';
import { MedicationsPlate } from './medications-plate';
import { MedicationsPlateSkeleton } from './medications-plate-skeleton';
import { MedicationsTitle } from './medications-title';

const mockType = MedicationsType.REGULAR;
const mockMedications = mockMedicationsWidgetData[mockType.toLocaleLowerCase()];

describe('MedicationsPlate', () => {
  let component: ReturnType<typeof shallow>;

  beforeEach(() => {
    component = shallow(
      <MedicationsPlate
        type={mockType}
        medications={mockMedications}
      />
    );
  });

  it('component should match snapshot', () => {
    expect(toJson(component)).toMatchSnapshot();
  });

  it('skeleton should match snapshot', () => {
    component = shallow(<MedicationsPlateSkeleton />);

    expect(toJson(component)).toMatchSnapshot();
  });

  it('should contain right count of medications', () => {
    expect(component.find(MedicationsTitle).props().count).toEqual(mockMedications.length);
    expect(component.find(MedicationsItem)).toHaveLength(mockMedications.length);
  });
});
