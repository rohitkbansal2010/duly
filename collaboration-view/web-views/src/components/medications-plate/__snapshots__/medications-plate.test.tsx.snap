// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MedicationsPlate component should match snapshot 1`] = `
<div
  className="medicationsPlateContainer"
>
  <MedicationsTitle
    count={5}
    type="REGULAR"
  />
  <MedicationsHeader
    type="REGULAR"
  />
  <div
    className="medicationsPlateMain"
  >
    <MedicationsItem
      id="Regular_Medication_Id4"
      instructions="Take 1 capsule by month 2 times a day"
      key="Regular_Medication_Id4"
      provider={
        Object {
          "familyName": "Reyes",
          "givenNames": Array [
            "Ana",
            "Maria",
          ],
          "prefixes": Array [
            "Dr.",
          ],
        }
      }
      reason="For potassium replacement (COPD)"
      scheduleType="Regular"
      title="APotassium chloride ER Capsule 10 mEq third"
    />
    <MedicationsItem
      id="Regular_Medication_Id3"
      instructions="Take 1 capsule by month 2 times a day"
      key="Regular_Medication_Id3"
      provider={
        Object {
          "familyName": "Reyes",
          "givenNames": Array [
            "Ana",
            "Maria",
          ],
          "prefixes": Array [
            "Dr.",
          ],
        }
      }
      reason="For potassium replacement (COPD)"
      scheduleType="Regular"
      title="BPotassium chloride ER Capsule 10 mEq fourth"
    />
    <MedicationsItem
      id="Regular_Medication_Id_2"
      key="Regular_Medication_Id_2"
      scheduleType="Regular"
      title="CPotassium chloride ER Capsule 10 mEq fifth"
    />
    <MedicationsItem
      id="Regular_Medication_Id2"
      instructions="Take 1 capsule by month 2 times a day"
      key="Regular_Medication_Id2"
      provider={
        Object {
          "familyName": "Reyes",
          "givenNames": Array [
            "Ana",
            "Maria",
          ],
          "prefixes": Array [
            "Dr.",
          ],
        }
      }
      reason="For potassium replacement (COPD)"
      scheduleType="Regular"
      startDate="2022-02-01T00:00:00Z"
      title="Potassium chloride ER Capsule 10 mEq first"
    />
    <MedicationsItem
      id="Regular_Medication_Id"
      instructions="Take 1 capsule by month 2 times a day"
      key="Regular_Medication_Id"
      provider={
        Object {
          "familyName": "Reyes",
          "givenNames": Array [
            "Ana",
            "Maria",
          ],
          "prefixes": Array [
            "Dr.",
          ],
        }
      }
      reason="For potassium replacement (COPD)"
      scheduleType="Regular"
      startDate="2022-02-01T00:00:00Z"
      title="Potassium chloride ER Capsule 10 mEq second"
    />
  </div>
</div>
`;

exports[`MedicationsPlate skeleton should match snapshot 1`] = `
<div
  className="medicationsPlateSkeleton"
>
  <DulyLoader
    width={23}
  />
</div>
`;
